oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConstants.java

```diff
@@ -57,9 +60,48 @@ public static String getConnectionString(String connectionType,
     }
     if (OracleConstants.SERVICE_CONNECTION_TYPE.equalsIgnoreCase(connectionType)) {
       return String.format(OracleConstants.ORACLE_CONNECTION_STRING_SERVICE_NAME_FORMAT,
-          host, port, database);
+             DEFAULT_CONNECTION_PROTOCOL, host, port, database);
     }
     return String.format(OracleConstants.ORACLE_CONNECTION_STRING_SID_FORMAT,
-        host, port, database);
+           DEFAULT_CONNECTION_PROTOCOL, host, port, database);
+  }
+
+  /**
+   * Constructs the Oracle connection string based on the provided connection type, host, port, and database.
+   * If SSL is enabled, the connection protocol will be "tcps" instead of "tcp".
+   *
+   * @param connectionType TNS/Service/SID
+   * @param host Host name of the oracle server
+   * @param port Port of the oracle server
+   * @param database Database to connect to
+   * @param useSSL Whether SSL/TLS is required(YES/NO)
+   * @return Connection String based on the given parameters and connection type.
+   */
+  public static String getConnectionString(String connectionType,
```

Comment:
I cannot see any reason of this new method, why can't we reuse the existing one?

Reply:
This was actually overloaded to reduce the impact of these changes on CDAP connectors, this method will only be used by Source and Sink plugin.

Oracle Action, Post Action and Oracle Connector will still use the old method to generate connection string.

Reply:
If we are using this method here: https://github.com/data-integrations/database-plugins/pull/515/files#diff-bd85394fe9ff353467fd45fd10ad329b508e0fbbc70145fe34de0d54f483a99bR129

It means it does impact CDAP connectors so it does not really help by overriding the method.

Reply:
got it, removed. [b68ec933ff27ff2a1ade2c578727113604803367]